File: minDepth.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def minDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if not root:
            return 0
        if not root.left and not root.right:
            return 1
        if not root.left or not root.right:
            return max(self.minDepth(root.left),self.minDepth(root.right))+1
        return min(self.minDepth(root.left),self.minDepth(root.right))+1

File: java/minDepth.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    
    public int minDepth(TreeNode root) {
        if(root==null){
            return 0;
        }
        else if((root.left!=null && root.right==null)||(root.left==null && root.right!=null)){
            return Math.max(minDepth(root.left),minDepth(root.right))+1;
        }

        return Math.min(minDepth(root.left),minDepth(root.right))+1;
        
    }
}